docs/src/main/sphinx/connector/faker.md

```diff
@@ -0,0 +1,160 @@
+# Faker connector
+
+The Faker connector generates realistically looking, random data matching a
+defined structure.
+
+## Configuration
+
+The connector can create data in Trino and allowing querying it. Create a
+catalog properties file that specifies the Faker connector by setting the
+`connector.name` to `faker`.
+
+For example, to access data as the `generator` catalog, create the file
+`etc/catalog/generator.properties`. Replace the connection properties as
+appropriate for your setup:
+
+
+```text
+connector.name=faker
+faker.null-probability=0.1
+faker.default-limit=1000
+```
+
+## Examples
+
+Faker generates data when reading from a table created in a catalog using this
+connector. This makes it easy to fill an existing schema with random data, by
+copying only the schema into a Faker catalog, and inserting the data back into
+the original tables.
+
+```sql
+CREATE TABLE faker.default.customer (LIKE production.public.customer EXCLUDING PROPERTIES);
+
+INSERT INTO production.public.customers
+SELECT *
+FROM faker.default.customers
+WHERE
+  born_at BETWEEN CURRENT_DATE - INTERVAL '150' YEAR AND CURRENT_DATE
+  AND age_years BETWEEN 0 AND 150
+LIMIT 100;
+```
+
+To generate more realistic data, choose specific generators by setting the
+`generator` property on columns:
+
+```sql
+SHOW CREATE TABLE production.public.customers;
+
+-- copy the output of the above query and add some properties:
+CREATE TABLE faker.default.customer (
+  id UUID NOT NULL,
+  name VARCHAR NOT NULL WITH (generator = '#{Name.first_name} #{Name.last_name}'),
+  address VARCHAR NOT NULL WITH (generator = '#{Address.fullAddress}'),
+  born_at DATE,
+  age_years INTEGER
+);
+```
+
+See the Datafaker's documentation for more information about
+[the expression](https://www.datafaker.net/documentation/expressions/) syntax
+and [available providers](https://www.datafaker.net/documentation/providers/).
+
+## Data generation
```

Comment:
To a large degree this whole data generation is actually part of configuration .. lets move it there.

Reply:
All the configuration is optional. I want to first show users the simplest use case. Maybe I can move creating the catalog from `Configuration` to `Usage`, keep the properties reference in `Configuration`, and then merge it with `Data generation`?

Reply:
I moved things around, PTAL